libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxPrimaryManeuver.kt

```diff
@@ -0,0 +1,159 @@
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ImageSpan
+import android.util.AttributeSet
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
+import com.mapbox.navigation.ui.base.MapboxView
+import com.mapbox.navigation.ui.base.model.maneuver.DelimiterComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.ExitNumberComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState
+import com.mapbox.navigation.ui.base.model.maneuver.PrimaryManeuver
+import com.mapbox.navigation.ui.base.model.maneuver.RoadShieldComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.TextComponentNode
+import com.mapbox.navigation.ui.maneuver.R
+
+/**
+ * Default view to render primary banner instructions onto [MapboxManeuverView].
+ * It can be directly used in any other layout.
+ * @property attrs AttributeSet
+ * @property defStyleAttr Int
+ */
+class MapboxPrimaryManeuver @JvmOverloads constructor(
+    context: Context,
+    private val attrs: AttributeSet? = null,
+    private val defStyleAttr: Int = 0
+) : MapboxView<ManeuverState>, AppCompatTextView(context, attrs, defStyleAttr) {
+
+    private var leftDrawable = ContextCompat.getDrawable(
+        context, R.drawable.mapbox_ic_exit_arrow_left
+    )
+    private var rightDrawable = ContextCompat.getDrawable(
+        context, R.drawable.mapbox_ic_exit_arrow_right
+    )
+    private var exitBackground = ContextCompat.getDrawable(
+        context, R.drawable.mapbox_exit_board_background
+    )
+
+    /**
+     * Entry point for [MapboxPrimaryManeuver] to render itself based on a [ManeuverState].
+     */
+    override fun render(state: ManeuverState) {
+        when (state) {
+            is ManeuverState.ManeuverPrimary.Instruction -> {
+                renderPrimaryManeuver(state.maneuver)
+            }
+            is ManeuverState.ManeuverPrimary.Show -> {
+                updateVisibility(VISIBLE)
+            }
+            is ManeuverState.ManeuverPrimary.Hide -> {
+                updateVisibility(GONE)
+            }
+            else -> {
+            }
+        }
+    }
+
+    private fun renderPrimaryManeuver(maneuver: PrimaryManeuver) {
+        val instruction = generateInstruction(maneuver, lineHeight)
+        if (instruction.isNotEmpty()) {
+            text = instruction
+        }
+    }
+
+    private fun updateVisibility(visibility: Int) {
+        this.visibility = visibility
+    }
+
+    private fun updateMaxLines(maxLines: Int) {
+        this.maxLines = maxLines
+    }
+
+    private fun generateInstruction(
+        maneuver: PrimaryManeuver,
+        desiredHeight: Int
+    ): SpannableStringBuilder {
+        val instructionBuilder = SpannableStringBuilder()
+        maneuver.componentList.forEach { component ->
+            when (component.node) {
```

Comment:
```suggestion
        maneuver.componentList.forEach { component ->
            val node = component.node
            when (node) {
```

nit, this will make Kotlin smart cast work